### DISCUSSION TITLE
Extracting embedded boundary data from plotfiles

### DISCUSSION THREAD

**Discussion by ashwathsv** [AMReX-Codes/amrex#3364] [Q&A]
I am trying to compute the forces on the surface of an embedded boundary using the cut-cell pressure values. This requires details of the normal and tangential vectors for the EB surface at each cut-cell. All the computations were performed with EBSupport:full option.

Does the AMReX repo have a script to extract embedded boundary data from the plotfiles to compute these vectors? 

Thanks in advance!

**Comment by WeiqunZhang** [comment]
Are you using AmrLevel? If so, you can add what you need to `derive_lst` and set `amr.derive_plot_vars = "name1 name2" in inputs. Note that this is for cell-centered variables only. If you need other types, you have to save them as raw VisMF files and read them.

**Comment by ashwathsv** [comment]
Yes, I am using AMRLevel. How can I access the MultiFabs and MultiCutFabs containing the EB data (area fractions, face and boundary centroids, boundary normals) inside one of the functions defined in CNS_derive.cpp? 

Currently, I am only able to access the EBCellFlag by the following statement: const auto& flag_fab = amrex::getEBCellFlagFab(datfab);
